src/main/anovos/data_analyzer/stats_generator.py

```diff
@@ -415,9 +414,11 @@ def unionAll(dfs):
         )
 
     odf = unionAll(list_df)
+
     if print_impact:
+        odf.persist()
```

Comment:
It is a good practice to always do `unpersist` after `persist`. Can you add that in here?